clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -254,28 +250,34 @@ public String toString() {
         }
     }
 
-    public static class FutureCompletionHandler implements RequestCompletionHandler {
-
-        private final CompletableFuture<ClientResponse> future;
+    public static class FutureCompletionHandler extends CompletableFuture<ClientResponse> implements RequestCompletionHandler {
```

Comment:
Why do you want to inherit from `CompletableFuture`? Extending base classes might lead to the fragile base class problem. I do not see that your code manifests this problem at the moment, but I think it would be better to minimize the surface for such problem if possible and if it does not cost too much.